### PR TITLE
fix(muyajs): allow CJK ideographs as flanking boundary for emphasis

### LINKED ISSUE #4307: [Bug] 似乎不能正常处理**加粗语法的配对
### Description

似乎不能正确处理文字加粗语法\**的配对，一对\**的内部有某些符号（如' " ( [ 等符号）时，写成  **“加粗” **时就不能正常渲染，但是写成\**加粗\**时正常。（Github论坛的处理也与下图相同）

`比如写成如：  例子例子**“加粗”**例子例子，例子例子例子例子例子例子例子**加粗**例子例子。渲染的结果不是想要的，`

如果删除第一个“加粗”左右两侧的引号，则能正确渲染。

### Steps to Reproduce

如上

### Expected Behavior

Image

### Actual Behavior

Image

### Screenshots / Logs

_No response_

### Operating System

Windows

### MarkText Version

0.19.0

### OS Version

windows 11

### Reproducibility

- [x] I can reliably reproduce this issue

### PR DESCRIPTION
**Pull request by Jocs**
## Summary

Closes #4307.

`**…**` adjacent to a CJK ideograph + inner punctuation (e.g. `中文**"加粗"**中文`, `日本語**(強調)**日本語`, `한국어**[강조]**한국어`) fails to render as strong in WYSIWYG. The same input has been broken since at least v0.17.1 (the parser code was byte-identical until this PR), and GitHub/GFM renders the same way.

## ⚠️ Standards compliance — this PR ships a non-standard extension

This is **not** a CommonMark conformance fix. It is a deliberate divergence from CommonMark spec 6.2 ("Emphasis and strong emphasis", https://spec.commonmark.org/0.31.2/#emphasis-and-strong-emphasis):

- Spec 6.2 only counts **Unicode whitespace** and **Unicode punctuation** as flanking boundaries.
- CJK Unified Ideographs are Unicode category `Lo` — neither whitespace nor punctuation.
- Under a literal reading of the spec, `中文**"加粗"**中文` therefore MUST NOT open a strong run. GitHub/GFM is strictly spec-conformant and behaves the same way.

CJK scripts do not use spaces between words, so the strict rule effectively denies emphasis to any CJK paragraph that wraps the `**` run with punctuation (quotes, parentheses, brackets, 、 …). This PR adopts the well-known CJK-friendly emphasis extension already shipped by Typora, VSCode markdownlint, Joplin, and most CJK-oriented Markdown tools: treat CJK ideographs as boundary-equivalent in clause (2b) of the flanking check.

**Why this is safe:**

- The widening is **additive only** (`||`): it never rejects emphasis CommonMark accepts, it only accepts emphasis CommonMark rejects on the CJK-boundary edge case.
- All ASCII / English inputs continue to parse byte-identically — verified by 3 regression cases in the new vitest and by the full 556-test unit suite remaining green.
- Documented in the source inline (`packages/muyajs/lib/parser/utils.js`) with explicit "NON-STANDARD EXTENSION" header so future readers / spec auditors can see the divergence at a glance.

## Implementation

- `packages/muyajs/lib/parser/utils.js` — introduce `CJK_REG` (CJK Unified Ideographs BMP + Ext-A + Ext-B via surrogate pairs + Compatibility Ideographs, Hiragana, Katakana, Halfwidth Katakana, Hangul Syllables) and extend the (2b) clauses of `canOpenEmphasis` / `canCloseEmphasis` with a three-way `||`.
- `packages/desktop/src/types/muya.d.ts` — small ambient shim for `muya/lib/parser` so the new vitest's named-import passes `vue-tsc`, consistent with the existing muya bridges.

## Tests

- `packages/desktop/test/unit/specs/markdown-strong-cjk.spec.ts` — **token-level** vitest: 3 CJK + inner-punctuation cases (fail on `develop`, pass with this PR) + 3 regression cases (already pass on `develop`, still pass).
- `packages/desktop/test/e2e/strong-cjk.spec.ts` — **DOM-level** Playwright spec: confirms `<strong>` is emitted in the live WYSIWYG render.

## Test plan

- [x] `pnpm test:unit` — 556/556 pass (10 files)
- [x] `pnpm -C packages/desktop exec playwright test test/e2e/` — 75 pass / 2 skip / 0 fail (incl. 2 new specs)
- [x] `pnpm run lint` — 0 errors
- [x] `pnpm run typecheck` — pass
- [x] Verified red-first: new specs fail on `develop` before the parser change, then pass after.
- [x] Manually verified the same inputs still fail in v0.17.1 (parser file is byte-identical to pre-fix develop), confirming this is a pre-existing bug, not a 0.19 regression.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/muyajs/lib/parser/utils.js [hunk 1]
```diff
@@ -124,10 +148,13 @@ const canOpenEmphasis = (src, marker, pending) => {
   // and either (2a) not followed by a punctuation character,
   // or (2b) followed by a punctuation character and preceded by Unicode whitespace or a punctuation character.
   // For purposes of this definition, the beginning and the end of the line count as Unicode whitespace.
-  if (PUNCTUATION_REG.test(followedChar) && !(UNICODE_WHITESPACE_REG.test(precededChar) || PUNCTUATION_REG.test(precededChar))) {
+  // Non-standard CJK widening (see CJK_REG block above) — additive only:
+  // CJK ideographs are accepted as "preceded by" boundary, on top of the
+  // CommonMark-defined whitespace/punctuation set.
+  if (PUNCTUATION_REG.test(followedChar) && !(UNICODE_WHITESPACE_REG.test(precededChar) || PUNCTUATION_REG.test(precededChar) || CJK_REG.test(precededChar))) {
     return false
```
[the remark this answers is not in the document]
Comment (Jocs) [reply packages/muyajs/lib/parser/utils.js:187]: Good catch — verified and addressed in f6a53aa5.

`charAt` / `src[i]` only return one UTF-16 code unit, so the surrogate-pair branch of `CJK_REG` (and, I now realize, the long list of surrogate-pair alternations in the existing CommonMark `PUNCTUATION_REG`) was effectively dead. Both are now hit because every flanking-boundary read goes through new `lastCodePointChar` / `codePointCharAt` helpers that return 1 or 2 code units as appropriate.

Added a vitest case `𠀀𠀁**"加粗"**𠀀𠀁` (CJK Ext-B, U+20000/U+20001) to lock the surrogate-pair behavior down. Full suite still green (557 unit / 75 e2e).
